src/include/lean/mpz.h

```diff
@@ -214,11 +214,11 @@ class mpz {
     void submul(mpz const & a, mpz const & b) { mpz_submul(m_val, a.m_val, b.m_val); }
 
     // a <- b * 2^k
-    friend void mul2k(mpz & a, mpz const & b, unsigned k) { mpz_mul_2exp(a.m_val, b.m_val, k); }
+    friend void mul2k(mpz & a, mpz const & b, unsigned long k) { mpz_mul_2exp(a.m_val, b.m_val, k); }
```

Comment:
Thanks -- that's unfortunate.  The underlying GMP type for a shift `mp_bitcnt_t`which is always an `unsigned long` if I'm reading the source correctly.  This would mean the shift amount is at most 2^32 on Windows.

How many bits can the result of `lean_unbox` contain?  We need mul2k to take a type at least large as that.

Reply:
The easiest fix would be to just have shift operations fail on values larger than 2^32 (perhaps only on Windows).  Those seem like they would be very unlikely to be actually needed.

Reply:
Yeah, restricting the RHS to < 2^32 seems fine to me on all platforms. Though perhaps we should make an exception for LHS 0? :)

Reply:
I've implemented this.  I also caught a bug in nat.pow